From e20bd64b75a98e1082a0fc1994580ccef1215cde Mon Sep 17 00:00:00 2001
From: Daniel Mueller <deso@posteo.net>
Date: Sun, 25 Nov 2018 21:53:41 -0800
Subject: [PATCH] Set /tmp/ as the default directory when importing .ics files

While other file imports in Thunderbird remember the last directory the
user navigated to, the .ics import code does not do so. That is very
annoying, especially when the default dialog starts off showing the
useless "Recent" items.
As for all our intents and purposes /tmp/ is the directory we want to
import from, this change adjusts the code to start up a dialog showing
the contents of this directory.
---
 comm/calendar/base/content/import-export.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/comm/calendar/base/content/import-export.js b/comm/calendar/base/content/import-export.js
index bf8c40..8d7dfb 100644
--- a/comm/calendar/base/content/import-export.js
+++ b/comm/calendar/base/content/import-export.js
@@ -24,6 +24,9 @@ function loadEventsFromFile(aCalendar) {
 
     let picker = Components.classes["@mozilla.org/filepicker;1"]
                            .createInstance(nsIFilePicker);
+    let uri = Services.io.newURI("file:///tmp/");
+    let dir = uri.QueryInterface(Components.interfaces.nsIFileURL).file;
+    picker.displayDirectory = dir;
     picker.init(window,
                 cal.l10n.getCalString("filepickerTitleImport"),
                 nsIFilePicker.modeOpenMultiple);
-- 
2.18.1

